### PR TITLE
feat: dnssec load keys from AWS Secrets Manager

### LINKED ISSUE #6616: Enhancement: Integrate AWS Secrets Manager for DNSSEC Key Management
**What would you like to be added**:
Integration of AWS Secrets Manager for DNSSEC key management within the CoreDNS DNSSEC plugin. This feature should enable the DNSSEC plugin to retrieve cryptographic keys directly from AWS Secrets Manager, moving away from the dependency on locally stored key files.

**Why is this needed**:
The current implementation of the DNSSEC plugin in CoreDNS requires cryptographic keys to be stored in local files. This approach can inadvertently lead to insecure practices, such as embedding keys in container images or including them in version control systems, to simplify deployment processes. Such practices expose keys to higher risks of unauthorized access and potential leakage.

Integrating AWS Secrets Manager can mitigate these risks by providing a secure and centralized key management solution, which offers:

1. **Secure Storage and Access**: Keys are stored securely in the cloud with encryption at rest and are accessed over encrypted channels, minimizing the risk of exposure during storage and transmission.

2. **Elimination of Insecure Practices**: By fetching keys directly from a managed service, there's no need to store keys within the application's operational environment or codebase. This reduces the temptation to use insecure shortcuts in key management during deployment.

3. **Best Practices**: Using a managed service like AWS Secrets Manager helps in adhering to security best practices.

This enhancement could improve the security of CoreDNS deployments by providing a more secure method to manage cryptographic keys, aligning with best practices for secure software deployment.

### PR DESCRIPTION
**Pull request by kcolemangt**
### 1. Why is this pull request needed and what does it do?
Enhancement: Integrate AWS Secrets Manager for DNSSEC Key Management

### 2. Which issues (if any) are related?
Closes #6616

### 3. Which documentation changes (if any) need to be made?
Included
### 4. Does this introduce a backward incompatible change or deprecation?
No

### CODE REVIEW

**Comment by kcolemangt** [comment]
The CI failure in `TestReadme` is expected. The added example demonstrates pulling a DNSSEC key from AWS Secrets Manager, which fails in the CI environment as it lacks the necessary AWS credentials. This example is intended solely for documentation purposes.

**Comment by kcolemangt** [comment]
> The CI failure in `TestReadme` is expected. The added example demonstrates pulling a DNSSEC key from AWS Secrets Manager, which fails in the CI environment as it lacks the necessary AWS credentials. This example is intended solely for documentation purposes.

@chrisohaver 👀🙏

File: plugin/dnssec/README.md [hunk 1]
```diff
@@ -49,6 +49,22 @@ used.
     * generated public key `Kexample.org+013+45330.key`
     * generated private key `Kexample.org+013+45330.private`
 
+* `key secret` indicates that **KEY** secret(s) should be read from AWS Secrets Manager. Secret names or
```
Comment (chrisohaver) [inline plugin/dnssec/README.md]: I think the option name should have `aws` in it, to indicate that it is a aws specific integration.  e.g. `aws_secrets` or `aws_secretsmanager` or something.

**Comment by chrisohaver** [comment]
> The CI failure in `TestReadme` is expected. The added example demonstrates pulling a DNSSEC key from AWS Secrets Manager, which fails in the CI environment as it lacks the necessary AWS credentials. This example is intended solely for documentation purposes.

There is a way to exclude example corefiles from being tested in that test.  I don't remember off the top of my head...

Speaking of which, I presume that the aws client library pulls the credentials from the env or something?

File: plugin/dnssec/README.md (hunk 1 above)
Comment (kcolemangt) [reply plugin/dnssec/README.md]: Good idea.

File: plugin/dnssec/README.md [hunk 2]
```diff
@@ -49,6 +49,26 @@ used.
...
+  '{key: $key, private: $private}')"
+  ```
+
+  This command reads the contents of the `.key` and `.private` files, constructs a JSON object, and stores it
+  as a new secret in AWS Secrets Manager with the specified name and description. CoreDNS will then fetch
+  the key data from AWS Secrets Manager when using the `key aws_secretsmanager` directive.
+
+  [AWS SDK for Go V2](https://aws.github.io/aws-sdk-go-v2/docs/configuring-sdk/#specifying-credentials) is used
+  for authentication with AWS Secrets Manager. Make sure the provided AWS credentials have the necessary
+  permissions (e.g., `secretsmanager:GetSecretValue`) to access the specified secrets in AWS Secrets Manager.
```
Comment (kcolemangt) [inline plugin/dnssec/README.md:70]: > Speaking of which, I presume that the aws client library pulls the credentials from the env or something?

@chrisohaver yes. Clarifying in the docs now.

**Comment by kcolemangt** [comment]
> There is a way to exclude example corefiles from being tested in that test. I don't remember off the top of my head...

@chrisohaver Thanks for pointing me in the right direction. Tests are built when the `corefile` language is specified in a code block.

**Review by chrisohaver — APPROVED** [review]

**Pull request merged**